src/client/opamAuxCommands.ml

```diff
@@ -386,6 +386,29 @@ let simulate_local_pinnings ?quiet ?(for_view=false) st to_pin =
            st.switch_global st.switch st.switch_config ~pinned
            ~opams:local_opams)
     );
+    reinstall = lazy (
+      let open OpamPackage.Set.Op in
+      let installed_pinned = st.pinned %% st.installed in
+      OpamPackage.Set.fold (fun pinned_pkg reinstall ->
+          match
+            OpamPackage.Set.find_opt
+              (fun pkg ->
+                 OpamPackage.Name.equal
+                   (OpamPackage.name pinned_pkg)
+                   (OpamPackage.name pkg))
+              local_packages
+          with
+          | None -> reinstall
+          | Some to_pin ->
```

Comment:
It is already pinned (simulate state), not `to_pin`

Reply:
good point. renamed to `local_pkg`